bee-message/src/output/unlock_condition/immutable_alias_address.rs

```diff
@@ -1,14 +1,17 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::address::AliasAddress;
+use crate::{
+    address::{Address, AliasAddress},
+    Error,
+};
 
 use derive_more::From;
 
 /// Defines the permanent [`AliasAddress`] that owns this output.
 #[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, From, packable::Packable)]
 #[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
-pub struct ImmutableAliasAddressUnlockCondition(AliasAddress);
+pub struct ImmutableAliasAddressUnlockCondition(#[packable(verify_with = verify_alias_address)] Address);
```

Comment:
If an `ImmutableAliasAddressUnlockCondition` needs an `AliasAddress` then maybe pass that instead of `Address`? More safety and less checks